src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -595,8 +578,9 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
         if (isResponseBodyEmpty) {
             // Empty response body, return the status code as error code back
             Logger.v(TAG, "Token request returned an empty response body.");
-            result = new AuthenticationResult(String.valueOf(statusCode), "", null);
+            throw new HttpResponseException(String.valueOf(statusCode));
         } else {
+            Logger.v(TAG, "Token request does not have exception.");
```

Comment:
This log entry doesn't make sense to me.  Was is it trying to communicate?  Should it just be removed?